From 5fac15319d55bf64d7d0c2f664aef0989187f7c4 Mon Sep 17 00:00:00 2001
From: Anna REITER <annareiter001@gmail.com>
Date: Wed, 26 Aug 2026 10:36:37 +0200
Subject: [PATCH 2/2] rvvts: DBT-RISE-RISCV: set vector width

Use VLEN 128 and ELEN 64 so MinresVP can run existing generated RVVTS
test sets for that vector configuration, e.g.
https://github.com/ics-jku/RVVTS_SailRV_Spike.

Signed-off-by: Anna REITER <annareiter001@gmail.com>
Signed-off-by: Manfred SCHLAEGL <manfred.schlaegl@gmx.at>
---
 src/iss/arch/rv32gcv.h | 4 ++--
 src/iss/arch/rv64gcv.h | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/iss/arch/rv32gcv.h b/src/iss/arch/rv32gcv.h
index 78b791a..9c633e3 100644
--- a/src/iss/arch/rv32gcv.h
+++ b/src/iss/arch/rv32gcv.h
@@ -330,7 +330,7 @@ template <> struct traits<rv32gcv> {
         RV_MIP_MEIP=2048ULL, RV_MIP_SGEIP=4096ULL, RV_MIP_LCOFIP=8192ULL, XLEN=32ULL,
         FLEN=64ULL, INSTR_ALIGNMENT=2ULL, RFS=32ULL, fence=0ULL,
         fencei=1ULL, fencevma=2ULL, CSR_SIZE=4096ULL, MUL_LEN=64ULL,
-        FFLAG_MASK=31ULL, VLEN=256ULL, ELEN=32ULL, EEW_MAX=32ULL,
+        FFLAG_MASK=31ULL, VLEN=128ULL, ELEN=64ULL, EEW_MAX=32ULL,
         FP_SEW_MIN=32ULL, FP_SEW_MAX=64ULL, VSTART_ADDR=8ULL, VXSAT_ADDR=9ULL,
         VXRM_ADDR=10ULL, VCSR_ADDR=15ULL, VL_ADDR=3104ULL, VTYPE_ADDR=3105ULL,
         VLENB_ADDR=3106ULL, MISA_VAL=1075843373ULL, MARCHID_VAL=0ULL, CLIC_NUM_IRQ=0ULL
@@ -338,7 +338,7 @@ template <> struct traits<rv32gcv> {
 
     
     constexpr static unsigned FP_REGS_SIZE = 64;
-    constexpr static unsigned V_REGS_SIZE = 256;
+    constexpr static unsigned V_REGS_SIZE = 128;
 
 
     enum reg_e {
diff --git a/src/iss/arch/rv64gcv.h b/src/iss/arch/rv64gcv.h
index 2be8f96..4a220cb 100644
--- a/src/iss/arch/rv64gcv.h
+++ b/src/iss/arch/rv64gcv.h
@@ -330,7 +330,7 @@ template <> struct traits<rv64gcv> {
         RV_MIP_MEIP=2048ULL, RV_MIP_SGEIP=4096ULL, RV_MIP_LCOFIP=8192ULL, XLEN=64ULL,
         FLEN=64ULL, INSTR_ALIGNMENT=2ULL, RFS=32ULL, fence=0ULL,
         fencei=1ULL, fencevma=2ULL, CSR_SIZE=4096ULL, MUL_LEN=128ULL,
-        FFLAG_MASK=31ULL, VLEN=512ULL, ELEN=64ULL, EEW_MAX=64ULL,
+        FFLAG_MASK=31ULL, VLEN=128ULL, ELEN=64ULL, EEW_MAX=64ULL,
         FP_SEW_MIN=32ULL, FP_SEW_MAX=64ULL, VSTART_ADDR=8ULL, VXSAT_ADDR=9ULL,
         VXRM_ADDR=10ULL, VCSR_ADDR=15ULL, VL_ADDR=3104ULL, VTYPE_ADDR=3105ULL,
         VLENB_ADDR=3106ULL, MISA_VAL=9223372036856877357ULL, MARCHID_VAL=0ULL, CLIC_NUM_IRQ=0ULL
@@ -338,7 +338,7 @@ template <> struct traits<rv64gcv> {
 
     
     constexpr static unsigned FP_REGS_SIZE = 64;
-    constexpr static unsigned V_REGS_SIZE = 512;
+    constexpr static unsigned V_REGS_SIZE = 128;
 
 
     enum reg_e {
